Assets/Plugins/Source/Core/Utility/FileUtility.cs

```diff
@@ -64,6 +64,73 @@ public static string GetProjectPath()
             return Path.GetFullPath(Path.Combine(Application.dataPath, "../"));
         }
 
+        /// <summary>
+        /// Generates a unique and new temporary directory inside the Temporary Cache Path as determined by Unity,
+        /// and returns the fully-qualified path to the newly created directory.
+        /// </summary>
+        /// <returns>Fully-qualified file path to the newly generated directory.</returns>
+        public static bool TryGetTempDirectory(out string path)
+        {
+            // Generate a temporary directory path.
+            string tempPath = Path.Combine(Application.temporaryCachePath, $"Output-{Guid.NewGuid()}/");
+
+            // If (by some crazy miracle) the directory path already exists, keep generating until there is a new one.
+            if (Directory.Exists(tempPath))
+            {
+                Debug.LogWarning(
+                    $"The temporary directory created collided with an existing temporary directory of the same name. This is very unlikely.");
+                tempPath = Path.Combine(Application.temporaryCachePath, $"Output-{Guid.NewGuid()}/");
+
+                if (Directory.Exists(tempPath))
+                {
+                    Debug.LogError(
+                        $"When generating a temporary directory, the temporary directory generated collided twice with already existing directories of the same name. This is very unlikely.");
+                    path = null;
+                    return false;
+                }
+            }
+
+            try
+            {
+                // Create the directory.
+                var dInfo = Directory.CreateDirectory(tempPath);
+
+                // Make sure the directory exists.
+                if (!dInfo.Exists)
+                {
+                    Debug.LogError($"Could not generate temporary directory.");
+                    path = null;
+                    return false;
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.LogError($"Could not generate temporary directory: {e.Message}");
+                path = null;
+                return false;
+            }
+
+            // return the fully-qualified path to the newly created directory.
+            path = Path.GetFullPath(tempPath);
+            return true;
+        }
+
+        #region Line Ending Manipulations
+
+        public static void ConvertDosToUnixLineEndings(string filename)
+        {
+            ConvertDosToUnixLineEndings(filename, filename);
+        }
+
+        public static void ConvertDosToUnixLineEndings(string srcFilename, string destFilename)
+        {
+            string fileContents = ReadAllText(srcFilename);
+            fileContents = fileContents.Replace("\r\n", "\n");
```

Comment:
This new method will end up with two copies of the string in memory, which is not as kind to the GC as the previous code.

Reply:
Ah that makes sense. This is only used while in the editor, and while copying files during the create package process, seems like a good trade-off for increased readability, but it's far from a hill that I'd die on 😄

Reply:
I would prefer to restore the previous functionality.

Reply:
Previous functionality has been restored.